merger/merger.go

```diff
@@ -107,21 +107,32 @@ const UnstableInsertIndexKey = "_gazelle_insert_index"
 // If a rule is marked with a "# keep" comment, the whole rule will not
 // be modified.
 func MergeFile(oldFile *rule.File, emptyRules, genRules []*rule.Rule, phase Phase, kinds map[string]rule.KindInfo) {
-	getMergeAttrs := func(r *rule.Rule) map[string]bool {
+	isMergeable := func(r *rule.Rule, attr string) bool {
+		// Attributes merged in this phase.
 		if phase == PreResolve {
-			return kinds[r.Kind()].MergeableAttrs
+			return kinds[r.Kind()].MergeableAttrs[attr]
 		} else {
-			return kinds[r.Kind()].ResolveAttrs
+			return kinds[r.Kind()].ResolveAttrs[attr]
 		}
 	}
 
+	isManaged := func(r *rule.Rule, attr string) bool {
+		// Name and visibility are uniquely managed by gazelle.
+		if attr == "name" || attr == "visibility" {
```

Comment:
I actually don't think `visibility` should be here but it causes too many changes in this repo atm. I think those changes are actually correct but they could be breaking such as reducing the visibility of `//internal/...`.